### PR TITLE
Migrate set-image.sh to go based test

### PR DESCRIPTION
**Pull request by ardaguclu**
`set-image.sh` tests in e2e-agnostic-ovn-cmd jobs are still [flaky](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_oc/1389/pull-ci-openshift-oc-master-e2e-agnostic-ovn-cmd/1644085811785764864), this still results in waste of time and resources.

There was an early attempt https://github.com/openshift/origin/pull/27046 to overcome this situation, but it was not merged due to requiring some tweaks. 

This PR makes changes on the previous attempt and migrates `set-image.sh` to `setimage.go` to run these tests also on different cluster variants for the better reliability.

### CODE REVIEW

**Comment by ardaguclu** [comment]
/hold
until the payloads are stabilized.

File: test/extended/cli/setimage.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-cli] oc set image", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		deploymentConfig = exutil.FixturePath("testdata", "cmd", "test", "cmd", "testdata", "test-deployment-config.yaml")
```
Comment (soltysh) [inline test/extended/cli/setimage.go:21]: Nit: use `exutil.FixturePath("testdata", "test-deployment-config.yaml")` instead, I'd like to get rid of that `testdata/cmd/` directory in the future entirely, so we should use files from `testdata` only.

File: test/extended/cli/setimage.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		imageStream      = exutil.FixturePath("testdata", "cmd", "test", "cmd", "testdata", "image-streams", "image-streams-centos7.json")
```
Comment (soltysh) [inline test/extended/cli/setimage.go:22]: Here, use `exutil.FixturePath("..", "..", "examples", "image-streams", "image-streams-centos7.json")`

**Pull request merged**